### ISSUE TITLE
n5 libblosc

### ISSUE THREAD

**Issue by tischi** [fiji/fiji#262]
related to: https://github.com/Blosc/c-blosc/issues/311

@axtimwalde @ctrueden  @constantinpape

This one seem so work for us at EMBL: https://oc.embl.de/index.php/s/MzyhCPM0FtzB5t7
If I put it into `Fiji.app/lib/linux64` and then run `new org.blosc.JBlosc()` I get `[WARNING] Ignoring unsupported output: result [org.blosc.JBlosc]` but none of the other error messages.
It comes from this path on our cluster environment: `/g/easybuild/x86_64/CentOS/7/skylake/software/Blosc/1.17.1-GCCcore-8.3.0/lib/libblosc.so.1.17.1`

**Comment by tischi** [comment]
@ctrueden  
I downloaded a fresh Fiji just now on a Mac and it did not contain the `libblosc` yet. 
On Linux however it does. Can you confirm this?

**Comment by constantinpape** [comment]
> If I put it into `Fiji.app/lib/linux64` and then run `new org.blosc.JBlosc()` I get `[WARNING] Ignoring unsupported output: result [org.blosc.JBlosc]` but none of the other error messages.
> It comes from this path on our cluster environment: `/g/easybuild/x86_64/CentOS/7/skylake/software/Blosc/1.17.1-GCCcore-8.3.0/lib/libblosc.so.1.17.1`

Just to give some context here: the EMBL cluster uses [easybuild](https://easybuild.readthedocs.io/en/latest/Introduction.html) as package manager. So this libblosc links to versions of libgcc and libstdc++ that are also provided by easybuild. 

Hence this is a possible fix to make FIJI run at EMBL (or a cluster systems with similar setup), but it doesn't help with the general problem of incompatible ABI versions.

**Comment by axtimwalde** [comment]
Can you please try this build?  https://files.gitter.im/550332cd15522ed4b3dd39fe/Cppo/libblosc.so

**Comment by tischi** [comment]
Lookin' good!

```
-bash-4.2$ /g/cba/tischer/software/Fiji.app/ImageJ-linux64 --headless --run /g/cba/tischer/software/tmp/blosc-test.groovy 
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option PermSize=128m; support was removed in 8.0
Java HotSpot(TM) 64-Bit Server VM warning: Using incremental CMS is deprecated and will likely be removed in a future release
[WARNING] Ignoring unsupported output: result [org.blosc.JBlosc]
```

**Comment by axtimwalde** [comment]
Excellent, uploading it to the updater now, note that this is 1.18.1, the last version before it became hard to include Snappy.  The CentOS shipped version that you tried earlier is of course much older because CentOS ;).  Still looking into statically linking libc and libstdc++...

**Comment by tischi** [comment]
Great! By the way: I think Fiji currently only ships the blosc lib for linux and windows, but still **not for mac**. I wrote this already somewhere else cc'ing Curtis, but I cannot find anymore where. So, if you by chance "bump into him", maybe you could remind him as well.

**Comment by ctrueden** [comment]
@tischi Thanks, I'm aware. It's on my list to investigate. I don't know what causes the issue though, so I have no estimate when it might be fixed.

**Comment by tischi** [comment]
@ctrueden thanks! let me know if it makes sense to consider some temporary workaround, e.g. adding the mac blosc lib to the MoBIE update site?!

**Comment by axtimwalde** [comment]
Great!  I filed a separate issue, so @ctrueden can collect solution badges ;).

**Comment by ctrueden** [comment]
It sounds like this issue should be closed in favor of #263? Please reopen if I misunderstood.
